save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/service/TestDiscoveringService.kt

```diff
@@ -20,79 +20,66 @@ import org.springframework.stereotype.Service
  * A service that call SAVE core to discover test suites and tests
  */
 @Service
-class TestDiscoveringService(private val configProperties: ConfigProperties) {
-    private val configDetector = ConfigDetector()
+class TestDiscoveringService {
+    /**
+     * Returns a root config of hierarchy; this config will already have all descendants merged with their parents.
+     *
+     * @param testResourcesRootAbsolutePath path to directory with root config
+     * @return a root [TestConfig]
+     */
+    @OptIn(ExperimentalFileSystem::class)
+    fun getRootTestConfig(testResourcesRootAbsolutePath: String): TestConfig =
+            ConfigDetector().configFromFile(testResourcesRootAbsolutePath.toPath()).apply {
+                getAllTestConfigs().onEach {
```

Comment:
Is it possible, that there is no any config file?

Reply:
Then `ConfigDetector().configFromFile` will throw `IllegalArgumentException` and execution will be marked as failed in `onErrorResume`. I'll add a note about `IAE` in this method's KDoc